buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -111,6 +111,11 @@ class PrecommitTasks {
         }
         task.dependsOn(project.sourceSets.test.classesTaskName)
         task.javaHome = project.runtimeJavaHome
+
+        // Java home name checking is fragile, but we control the environment
+        if (project.runtimeJavaHome.contains("fips")){
+            task.enabled = false
```

Comment:
I think it's worth to have a comment here explaining why we can't have jar hell in fips.